modules/sdk/gradle-plugins-jsdoc/src/main/java/com/liferay/gradle/plugins/jsdoc/AppJSDocPlugin.java

```diff
@@ -0,0 +1,185 @@
+
+	public static final String PLUGIN_NAME = "appJSDocConfiguration";
+
+	@Override
+	public void apply(Project project) {
+		GradleUtil.applyPlugin(project, NodePlugin.class);
+
+		final AppJSDocConfigurationExtension appJSDocConfigurationExtension =
+			GradleUtil.addExtension(
+				project, PLUGIN_NAME, AppJSDocConfigurationExtension.class);
+
+		final JSDocTask appJSDocTask = _addTaskAppJSDoc(project);
+
+		_addTaskJarAppJSDoc(appJSDocTask);
+
+		_configureTaskAppJSDoc(appJSDocTask, _addTaskDownloadAppJSDoc(project));
+
+		Gradle gradle = project.getGradle();
+
+		gradle.afterProject(
+			new Closure<Void>(project) {
+
+				@SuppressWarnings("unused")
+				public void doCall(Project subproject) {
+					Set<Project> subprojects =
+						appJSDocConfigurationExtension.getSubprojects();
+
+					if (subprojects.contains(subproject) &&
+						_hasJSDocSource(subproject)) {
+
+						_configureTaskAppJSDoc(appJSDocTask, subproject);
+					}
+				}
+
+			});
+	}
+
+	private JSDocTask _addTaskAppJSDoc(Project project) {
+		final JSDocTask jsDocTask = GradleUtil.addTask(
+			project, APP_JSDOC_TASK_NAME, JSDocTask.class);
+
+		jsDocTask.setDescription(
+			"Generates the API documentation for the JavaScript code in this " +
+				"app.");
+
+		jsDocTask.setDestinationDir(
+			new Callable<File>() {
+
+				@Override
+				public File call() throws Exception {
+					Project project = jsDocTask.getProject();
+
+					return new File(project.getBuildDir(), "jsdoc");
+				}
+
+			});
+
+		jsDocTask.setGroup(JavaBasePlugin.DOCUMENTATION_GROUP);
+
+		return jsDocTask;
+	}
+
+	private DownloadNodeModuleTask _addTaskDownloadAppJSDoc(Project project) {
+		DownloadNodeModuleTask downloadNodeModuleTask = GradleUtil.addTask(
+			project, DOWNLOAD_APP_JSDOC_TASK_NAME,
+			DownloadNodeModuleTask.class);
+
+		downloadNodeModuleTask.args("--no-save");
+		downloadNodeModuleTask.setDescription("Downloads JSDoc.");
+		downloadNodeModuleTask.setModuleName("jsdoc");
+		downloadNodeModuleTask.setModuleVersion(_VERSION);
+
+		return downloadNodeModuleTask;
+	}
+
+	private Jar _addTaskJarAppJSDoc(JSDocTask jsDocTask) {
+		Jar jar = GradleUtil.addTask(
+			jsDocTask.getProject(), JAR_APP_JSDOC_TASK_NAME, Jar.class);
+
+		jar.from(jsDocTask);
+		jar.setClassifier("jsdoc");
+		jar.setDescription(
+			"Assembles a jar archive containing the JavaScript documentation " +
+				"files for this app.");
+		jar.setGroup(BasePlugin.BUILD_GROUP);
+
+		return jar;
+	}
+
+	private void _configureTaskAppJSDoc(
+		JSDocTask appJSDocTask, DownloadNodeModuleTask downloadAppJSDocTask) {
+
+		appJSDocTask.dependsOn(downloadAppJSDocTask);
+	}
+
+	private void _configureTaskAppJSDoc(
+		JSDocTask appJSDocTask, Project subproject) {
+
+		JSDocTask subprojectJSDocTask = (JSDocTask)GradleUtil.getTask(
+			subproject, JSDocPlugin.JSDOC_TASK_NAME);
+
+		appJSDocTask.dependsOn(subprojectJSDocTask);
+
+		appJSDocTask.sourcesDirs(subprojectJSDocTask.getScriptFile());
```

Comment:
You don't need to pass the `scriptFile`: `appJSDoc` already depends on `downloadJSDoc` so it'll already point to the JSDoc script.